core/src/event.rs

```diff
@@ -1,19 +1,38 @@
 //! Iroha is a quite dynamic system so many events can happen.
 //! This module contains descriptions of such an events and
-//! utilitary Iroha Special Instructions to work with them.
+//! utilitarian Iroha Special Instructions to work with them.
 
-use eyre::{eyre, Result};
 use iroha_data_model::events::prelude::*;
+use iroha_macro::error::ErrorTryFromEnum;
 use tokio::sync::broadcast;
 use warp::ws::WebSocket;
 
-use crate::stream::{Sink, Stream};
+use crate::stream::{self, Sink, Stream};
 
 /// Type of `Sender<Event>` which should be used for channels of `Event` messages.
 pub type EventsSender = broadcast::Sender<Event>;
 /// Type of `Receiver<Event>` which should be used for channels of `Event` messages.
 pub type EventsReceiver = broadcast::Receiver<Event>;
 
+/// Type of error for `Consumer`
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    /// Error from provided stream/websocket
+    #[error("Stream error: {0}")]
+    Stream(#[from] stream::Error<<WebSocket as Stream<VersionedEventSubscriberMessage>>::Err>),
+
```

Comment:
Minor point. Our convention is to have no blank spaces between error variants. 

But this is a suggestion, not a requirement.